src/runtime_src/core/common/core_device.h

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (C) 2019 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CORE_DEVICE_H
+#define CORE_DEVICE_H
+
+// Please keep eternal include file dependencies to a minimum
+#include <boost/property_tree/ptree.hpp>
+#include <stdint.h>
+
+/**
+ * This class is a library abstract class where:
+ * 1) The header file is defined in core/common
+ * 2) The body is implemented in a lower SSA directory. 
+ *  
+ * Note 
+ *    The lower SSA directory for the given target (e.g., 
+ *    pcie/linux, pcie/windows, edge/linux) is responsible for
+ *    implementing the body and producing a xrt_core and
+ *    xrt_core_static library.
+ */
+namespace xrt_core {
+namespace device {
+  void get_devices(boost::property_tree::ptree &_pt);
+  void get_device_pcie_info(uint64_t _deviceID, boost::property_tree::ptree &_pt);
+  void get_device_rom_info(uint64_t _deviceID, boost::property_tree::ptree & _pt);
+  void get_device_xmc_info(uint64_t _deviceID, boost::property_tree::ptree & _pt);
+  void get_device_platform_info(uint64_t _deviceID, boost::property_tree::ptree & _pt);
+}
```

Comment:
It's not clear how do we decide what info goes into which API here. E.g., the serial number (implemented below in core_device.cpp)  is obtained from get_device_xmc_info(), but that seems to be part of platform info here? It seems like we go by subdevices (rom/xmc)? I'm not sure if this is ideal since we don't have these subdevices on Windows... Maybe we can have a chat?

Reply:
Hello Max, 
As you are aware, there are a lot of moving parts here and I did an initial "raw" guesstimate on how this information might be gathered.   By all means, if you have an idea on how this data should be organized, lets do chat so that we can organize it appropriately. 

Note
Being that for xbutil2, there is both refactoring and replacement coding happening at the same time, I am expecting that we will be re-examining methods/functions several times throughout the release.  The good news is that the new code is rather modular so making these changes are really simple and easy.  So keep on questioning and asking.

Reply:
OK, understood. It just seems that we don't want to tie this interface with subdevice which is implementation details. Yes, feel free to merge for now and we can chat along the way.